### ISSUE TITLE
When setting tolerations on workspace pods, each toleration appears multiple times in the pod spec

### ISSUE THREAD

**Issue by cccs-tom** [eclipse-che/che#19975]
### Describe the bug

When setting tolerations for the workspace pods as in the example below, the toleration spec appears multiple times on the pod.
```yaml
che:
  cheWorkspacePodTolerations:
    - key: "node.taint"
      operator: "Equal"
      value: "che-workspaces"
      effect: "NoExecute"
```

### Che version

 - [ ] latest
 - [ ] nightly
 - [x] other: please specify
7.30.1

### Steps to reproduce

1. In values.yaml, set che.cheWorkspacePodTolerations as in the example above.
2. Deploy to a Kubernetes cluster.
3. Use `kubectl get pod -oyaml` to retrieve the Pod spec for the workspace pod.
4. The specified Toleration will appear multiple times

### Expected behavior

Each unique toleration should appear only once in the Pod spec.

### Runtime
   - [x] kubernetes (include output of `kubectl version`)
   - [ ] Openshift (include output of `oc version`)
   - [ ] minikube (include output of `minikube version` and `kubectl version`)
   - [ ] minishift (include output of `minishift version` and `oc version`)
   - [ ] docker-desktop + K8S (include output of `docker version` and `kubectl version`)
   - [ ] other: (please specify)

Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.7", GitCommit:"6b3f9b283463c1d5a2455df301182805e65c7145", GitTreeState:"clean", BuildDate:"2021-05-19T22:28:47Z", GoVersion:"go1.15.12", Compiler:"gc", Platform:"linux/amd64"}

### Installation method
   - [ ] chectl
     * provide a full command that was used to deploy Eclipse Che (including the output)
     * provide an output of `chectl version` command
   - [ ] OperatorHub
   - [ ] I don't know

### Environment
   - [ ] my computer
       - [ ] Windows
       - [ ] Linux
       - [ ] macOS
   - [x] Cloud
       - [ ] Amazon
       - [x] Azure
       - [ ] GCE
       - [ ] other (please specify)
   - [ ] Dev Sandbox (workspaces.openshift.com)
   - [ ] other: please specify

**Comment by sympatheticmoose** [comment]
@skabashnyuk assume we can close this issue?

**Comment by benoitf** [comment]
@cccs-tom you may add in PR the prefix `fix #issuenumber` or `fixes` 
on bugs then the referenced issue is automatically closed when merged.

Core team usually also provide another PR to che-docs repository about new features
